asv_bench/benchmarks/dataset_io.py

```diff
@@ -164,3 +164,167 @@ def time_load_dataset_scipy_with_time_chunks(self):
+                                           frac_nan=0.2),
+                                     coords={'lon': lons, 'lat': lats,
+                                             'time': times},
+                                     dims=('time', 'lon', 'lat'),
+                                     name='foo', encoding=None,
+                                     attrs={'units': 'foo units',
+                                            'description': 'a description'})
+            ds['bar'] = xr.DataArray(randn((nt, self.nx, self.ny),
+                                           frac_nan=0.2),
+                                     coords={'lon': lons, 'lat': lats,
+                                             'time': times},
+                                     dims=('time', 'lon', 'lat'),
+                                     name='bar', encoding=None,
+                                     attrs={'units': 'bar units',
+                                            'description': 'a description'})
+            ds['baz'] = xr.DataArray(randn((self.nx, self.ny),
+                                           frac_nan=0.2).astype(np.float32),
+                                     coords={'lon': lons, 'lat': lats},
+                                     dims=('lon', 'lat'),
+                                     name='baz', encoding=None,
+                                     attrs={'units': 'baz units',
+                                            'description': 'a description'})
+
+            ds.attrs = {'history': 'created for xarray benchmarking'}
+
+            self.ds_list.append(ds)
+            self.filenames_list.append('test_netcdf_%i.nc' % i)
+
+
+class IOWriteMultipleNetCDF3(IOMultipleNetCDF):
+    def setup(self):
+        self.make_ds()
+        self.format = 'NETCDF3_64BIT'
+
+    def time_write_dataset_netcdf4(self):
+        xr.save_mfdataset(self.ds_list, self.filenames_list,
+                          engine='netcdf4',
+                          format=self.format)
+
+    def time_write_dataset_scipy(self):
+        xr.save_mfdataset(self.ds_list, self.filenames_list,
+                          engine='scipy',
+                          format=self.format)
+
+
+class IOReadMultipleNetCDF4(IOMultipleNetCDF):
+    def setup(self):
+
+        requires_dask()
+
+        self.make_ds()
+        self.format = 'NETCDF4'
+        xr.save_mfdataset(self.ds_list, self.filenames_list,
+                          format=self.format)
+
+    def time_load_dataset_netcdf4(self):
+        xr.open_mfdataset(self.filenames_list, engine='netcdf4').load()
+
+
+class IOReadMultipleNetCDF3(IOReadMultipleNetCDF4):
+    def setup(self):
+
+        requires_dask()
+
+        self.make_ds()
+        self.format = 'NETCDF3_64BIT'
+        xr.save_mfdataset(self.ds_list, self.filenames_list,
+                          format=self.format)
+
+    def time_load_dataset_scipy(self):
+        xr.open_mfdataset(self.filenames_list, engine='scipy').load()
```

Comment:
I would create a separate function just for the open_mfdataset call, without load, i.e.
```python
def time_open_dataset_scipy(self):
    ds = xr.open_mfdataset(self.filenames_list, engine='scipy')
```
This would allow us to track the cost of reading the indices and aligning. (This is my personal bottleneck with big datasets.)

Then `load` could be called in a different function.